With -Werror, deprecation warnings fail the build: these aren't critical in our
context, so ignore them.
---
diff --git a/build-logic/jvm/src/main/kotlin/gradlebuild.strict-compile.gradle.kts b/build-logic/jvm/src/main/kotlin/gradlebuild.strict-compile.gradle.kts
index f0333160..2ab90829 100644
--- a/build-logic/jvm/src/main/kotlin/gradlebuild.strict-compile.gradle.kts
+++ b/build-logic/jvm/src/main/kotlin/gradlebuild.strict-compile.gradle.kts
@@ -18,7 +18,7 @@ import gradlebuild.jvm.extension.StrictCompileExtension
 
 extensions.create<StrictCompileExtension>("strictCompile")
 
-val strictCompilerArgs = listOf("-Werror", "-Xlint:all", "-Xlint:-options", "-Xlint:-serial", "-Xlint:-classfile", "-Xlint:-try")
+val strictCompilerArgs = listOf("-Werror", "-Xlint:all", "-Xlint:-options", "-Xlint:-serial", "-Xlint:-classfile", "-Xlint:-try", "-Xlint:-deprecation")
 
 tasks.withType<JavaCompile>().configureEach {
     // Generated classes may not adhere to the strict no-warning policy that we apply to handwritten code
